Analysis/include/EventInfo.h

```diff
@@ -216,32 +216,29 @@ class EventInfoBase {
     {
         double bjet_pt_cut = period == analysis::Period::Run2017 ? cuts::btag_2017::pt : cuts::btag_2016::pt;
         double bjet_eta_cut = period == analysis::Period::Run2017 ? cuts::btag_2017::eta : cuts::btag_2016::eta;
-        double vbf_pt_cut = 30;
-        double vbf_eta_cut = 5;
         BTagger bTagger(period,jet_ordering);
 
         SelectedSignalJets selected_signal_jets;
 
-        const auto CreateJetInfo = [&]() -> auto {
+        const auto CreateJetInfo = [&](bool useBTag) -> auto {
             std::vector<analysis::jet_ordering::JetInfo<decltype(event.jets_p4)::value_type>> jet_info_vector;
             for(size_t n = 0; n < event.jets_p4.size(); ++n) {
                 if(selected_signal_jets.isSelectedBjet(n)) continue;
                 if(selected_signal_jets.isSelectedVBFjet(n)) continue;
 
                 if(period ==  analysis::Period::Run2017){
                     const double abs_eta = std::abs(event.jets_p4.at(n).eta());
-                    if(event.jets_p4.at(n).pt() < cuts::hh_bbtautau_2017::jetID::pt &&
-                        abs_eta > cuts::hh_bbtautau_2017::jetID::eta_low &&
-                        abs_eta < cuts::hh_bbtautau_2017::jetID::eta_high) continue;
-                    }
-
-                double tag = bTagger.BTag(event,n);
+                    if(event.jets_p4.at(n).pt() < cuts::hh_bbtautau_2017::jetID::max_pt_veto &&
```

Comment:
extract the veto code into a separate static method and use it to filter jets also inside SelectJets and GetHT methods.